btc_light_client_store.go

```diff
@@ -1,45 +1,59 @@
 package main
 
 import (
+	// "math/big"
+
 	"math/big"
 
 	"github.com/btcsuite/btcd/chaincfg/chainhash"
 	"github.com/btcsuite/btcd/wire"
 )
 
+const MaxForkAge = 12
+
 type Store interface {
 	LightBlockAtHeight(int64) *LightBlock
 	LatestHeight() int64
-	LatestLightBlock() *LightBlock
-	SetHeader(height int64, header wire.BlockHeader) error
-	CurrentTotalWork() *big.Int
 	LightBlockByHash(hash chainhash.Hash) *LightBlock
+	// check hash h is hash of latest block in remind fork sets.
+	RemindFork(h chainhash.Hash) bool
+	LatestCheckPoint() *LightBlock
+	AddBlock(parent *LightBlock, header wire.BlockHeader) error
+	SetLatestBlockOnFork(bh chainhash.Hash, latest bool) error
+	TotalWorkAtBlock(bh chainhash.Hash) *big.Int
+	SetBlock(lb *LightBlock, perviousPower *big.Int)
+	SetLatestCheckPoint(lb *LightBlock)
+	SetLightBlockByHeight(lb *LightBlock)
 }
 
 type MemStore struct {
-	latestHeight        int64
-	lightblockMap       map[int64]*LightBlock
-	lightBlockByHashMap map[chainhash.Hash]*LightBlock
+	lightblockMap         map[int64]*LightBlock
+	lightBlockByHashMap   map[chainhash.Hash]*LightBlock
+	latestBlockHashOfFork map[chainhash.Hash]struct{}
+	totalWorkMap          map[chainhash.Hash]*big.Int
+	latestcheckpoint      *LightBlock
 }
 
 func NewMemStore() *MemStore {
 	return &MemStore{
-		latestHeight:        0,
-		lightblockMap:       make(map[int64]*LightBlock),
-		lightBlockByHashMap: make(map[chainhash.Hash]*LightBlock),
+		lightblockMap:         make(map[int64]*LightBlock),
+		lightBlockByHashMap:   make(map[chainhash.Hash]*LightBlock),
+		latestBlockHashOfFork: make(map[chainhash.Hash]struct{}),
+		totalWorkMap:          make(map[chainhash.Hash]*big.Int),
+		latestcheckpoint:      nil,
 	}
 }
 
+func (s *MemStore) SetLightBlockByHeight(lb *LightBlock) {
+	s.lightblockMap[int64(lb.Height)] = lb
+}
+
 func (s *MemStore) LightBlockAtHeight(height int64) *LightBlock {
 	return s.lightblockMap[height]
 }
 
 func (s *MemStore) LatestHeight() int64 {
```

Comment:
Seem likes LatestHeight name is no sense since this function is get height of the latest checkpoint or latest finalize block.